components/sidebar/sidebar_channel_list/sidebar_channel_list.tsx

```diff
@@ -410,7 +409,44 @@ export default class SidebarCategoryList extends React.PureComponent<Props, Stat
 
     render() {
         const {categories} = this.props;
-        const renderedCategories = categories.map(this.renderCategory);
+
+        let channelList: React.ReactNode;
+        if (this.props.isUnreadFilterEnabled) {
```

Comment:
Swapping `UnreadChannels` for a set of `ChannelCategory` components gives us things like the `More Unreads` indicators for free as long as we still set the refs for each channel correctly.